openassessment/xblock/team_mixin.py

```diff
@@ -99,6 +104,41 @@ def valid_access_to_team_assessment(self):
         """
         return self.is_course_staff or self.in_studio_preview or self.has_team()
 
+    def add_team_submission_context(
+        self, context, team_submission_uuid=None, individual_submission_uuid=None, transform_usernames=False
+    ):
+        """
+        Adds team submission information to context dictionary, based on existing team submissions
+        Specifically team name and team_usernames
+
+        Params:
+            - context (dict): a context dict for rendering a page that we will add values to
+            - team_submission_uuid (string): [optional] the uuid of the team submission we want to add context info for
+            - individual_submission_uuid (string): [optional] the uuid of an individual submission that's a part of
+                                                   the team submission for which we want to add context info
+            - transform_usernames (bool): [optional default: False] If False, context['team_usernames'] will be a list
+                                          of username strings. If True, it will be a string, in the form
+                                          "Username1, Username2, ... UsernameN, and UsernameN+1"
+
+        One of team_submission_uuid and individual_submission_uuid are required, and if they are both provided,
+        individual_submission_uuid will be ignored.
+        """
+        if not any((team_submission_uuid, individual_submission_uuid)):
+            raise TypeError("One of team_submission_uuid or individual_submission_uuid must be provided")
+        if team_submission_uuid:
+            team_submission = get_team_submission(team_submission_uuid)
+        elif individual_submission_uuid:
+            team_submission = get_team_submission_from_individual_submission(individual_submission_uuid)
+
+        team = self.teams_service.get_team_by_team_id(team_submission['team_id'])
+        context['team_name'] = team.name
+
+        student_ids = get_team_submission_student_ids(team_submission['team_submission_uuid'])
+        usernames = [self.get_username(student_id) for student_id in student_ids]
+        if transform_usernames:
+            usernames = list_to_conversational_format(usernames)
```

Comment:
Unrelated rant: I still think `list_to_conversational_format` should be handled in the fronted or, minimally, a template filter. Maybe it shouldn't but it irks me to build display data in the backend.

Reply:
When we get react, it won't be.   Problem is the templates

Reply:
Somewhat agreed, but a ticket for another day